bindings/go/generator/jsonschemagen/extractor_test.go

```diff
@@ -0,0 +1,86 @@
+package jsonschemagen
+
+import (
+	"go/ast"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestExtractStructDocPrefersTypeSpecDoc(t *testing.T) {
+	ts := &ast.TypeSpec{
+		Name: ast.NewIdent("Example"),
+		Doc: commentGroup(
+			"// +marker",
+			"// nolint:revive",
+			"// Primary description",
+			"//",
+			"// Deprecated: use another type",
+			"// trailing note",
```

Comment:
Because I was seeing it earlier in this PR. What about "comment" lines, e.g. `/////////////////////////////`. `normalizeCommentLine` only trims `//` if I am not mistaken.

Reply:
added a test for this and made the parser stricter